docsite/rst/playbooks_tests.rst

```diff
@@ -0,0 +1,165 @@
+Jinja2 tests
+============
+
+.. contents:: Topics
+
+
+Test in Jinja2 are a way of evaluating template expressions and returning True or False.
+Jinja2 ships with many of these. See `builtin tests`_ in the official Jinja2 template documentation.
+Tests are very similar to filters and are used mostly the same way, but they can also be used in list
+processing filters like C(map()) and C(select()) to choose items in the list.
+
+Just like filters, tests always execute on the ansible controller, NOT on the target of a task, they test local data.
+
+In addition to those, Ansible supplies a few more, also users can easily create their own.
+
+.. _testing_strings:
+
+Testing strings
+---------------
+
+To match strings against a substring or a regex, use the "match" or "search" filter::
+
+    vars:
+      url: "http://example.com/users/foo/resources/bar"
+
+    tasks:
+        - shell: "msg='matched pattern 1'"
+          when: url | match("http://example.com/users/.*/resources/.*")
+
+        - debug: "msg='matched pattern 2'"
+          when: url | search("/users/.*/resources/.*")
+
+        - debug: "msg='matched pattern 3'"
+          when: url | search("/users/")
+
+'match' will require a complete match in the string, while 'search' will require a match inside of the string.
+
+
+.. _testing_versions:
+
+Version Comparison
+------------------
+
+.. versionadded:: 1.6
+
+To compare a version number, such as checking if the ``ansible_distribution_version``
+version is greater than or equal to '12.04', you can use the ``version_compare`` filter.
+
+The ``version_compare`` filter can also be used to evaluate the ``ansible_distribution_version``::
+
+    {{ ansible_distribution_version | version_compare('12.04', '>=') }}
+
+If ``ansible_distribution_version`` is greater than or equal to 12, this filter will return True, otherwise it will return False.
+
+The ``version_compare`` filter accepts the following operators::
+
+    <, lt, <=, le, >, gt, >=, ge, ==, =, eq, !=, <>, ne
+
+This test also accepts a 3rd parameter, ``strict`` which defines if strict version parsing should
+be used.  The default is ``False``, and if set as ``True`` will use more strict version parsing::
```

Comment:
The default is ``False``, but setting this as ``True`` uses more strict version parsing::